front/types/assistant/actions/retrieval.ts

```diff
@@ -39,13 +49,18 @@ export type TemplatedQuery = {
 // `query` and `relativeTimeFrame` will be used to generate the inputs specification for the model
 // in charge of generating the action inputs. The results will be used along with `topK` and
 // `dataSources` to query the data.
+export type RetrievalTimeframe = "auto" | "none" | TimeFrame;
+export type RetrievalQuery = "auto" | "none" | TemplatedQuery;
+export type RetrievalDataSourcesConfiguration =
```

Comment:
please no, I won't know it's an array by reading the code it adds an uncessary layer of typing.